pkg/ctl/utils/update_cluster_endpoint_access.go

```diff
@@ -85,17 +72,16 @@ func doUpdateClusterEndpoints(cmd *cmdutils.Cmd, newPrivate bool, newPublic bool
 	logger.Info("current Kubernetes API endpoint access: privateAccess=%v, publicAccess=%v",
 		curPrivate, curPublic)
 
-	if cfg.VPC.ClusterEndpoints.PublicAccess != nil && cfg.VPC.ClusterEndpoints.PrivateAccess != nil {
-		newPublic = *cfg.VPC.ClusterEndpoints.PublicAccess
-		newPrivate = *cfg.VPC.ClusterEndpoints.PrivateAccess
-	} else {
-		privateSet, publicSet := accessFlagsSet(cmd)
-		if !privateSet {
+	if argumentFlagsProvided(cmd) {
+		if !flagSet("private-access", cmd) {
 			newPrivate = curPrivate
 		}
-		if !publicSet {
+		if !flagSet("public-access", cmd) {
 			newPublic = curPublic
 		}
+	} else {
+		newPublic = *cfg.VPC.ClusterEndpoints.PublicAccess
+		newPrivate = *cfg.VPC.ClusterEndpoints.PrivateAccess
```

Comment:
If a ClusterConfig file is supplied with a blank `vpc.clusterEndpoints` field, the defaults set in `ClusterEndpointAccessDefaults()` can override the current settings.

Reply:
You are right. Good catch! I'll redo this part as I did initially by changing the loader and moving the validation to the loader instead of having it in `NewCtl()`